src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/SortKeyInterop.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Runtime.Versioning;
+
+namespace System.DirectoryServices.Protocols
+{
+    // Declared as partial in order to be able to set the different StructLayout
+    // attributes in the Windows and Linux specific files.
+    internal partial struct SortKeyInterop
+    {
+        public SortKeyInterop(SortKey sortKey)
+        {
+            if (sortKey == null)
+                throw new ArgumentNullException(nameof(sortKey));
+
+            AttributeName = sortKey.AttributeName;
+            MatchingRule = sortKey.MatchingRule;
+            ReverseOrder = sortKey.ReverseOrder;
+        }
+
+        public string AttributeName { get; set; }
+
+        public string MatchingRule { get; set; }
+
+        public bool ReverseOrder { get; set; }
```

Comment:
Personally, I wouldn't use auto-properties with a layout-controlled struct.  But, since they're not being mixed between declared fields and auto-declared fields, it's probably fine.  Maybe just a comment here that the properties can't be reordered because the struct has a controlled layout.

Reply:
(I'd probably just declare them all as `internal`-scoped fields and be done with it)

Reply:
Done, thanks!